custom_components/vicare/switch.py

```diff
@@ -0,0 +1,211 @@
+
+
+@dataclass
+class ViCareSwitchEntityDescription(SwitchEntityDescription, ViCareRequiredKeysMixin, ViCareToggleKeysMixin):
+    """Describes ViCare switch entity."""
+
+
+SWITCH_DESCRIPTIONS: tuple[ViCareSwitchEntityDescription, ...] = (
+    ViCareSwitchEntityDescription(
+        key=SWITCH_DHW_ONETIME_CHARGE,
+        name="Activate one-time charge",
+        icon="mdi:shower-head",
+        entity_category=EntityCategory.CONFIG,
+        value_getter=lambda api: api.getOneTimeCharge(),
+        enabler=lambda api: api.activateOneTimeCharge(),
+        disabler=lambda api: api.deactivateOneTimeCharge(),
+    ),
+)
+
+
+def _build_entity(name, vicare_api, device_config, description):
+    """Create a ViCare switch entity."""
+    _LOGGER.debug("Found device %s", name)
+    try:
+        description.value_getter(vicare_api)
+        _LOGGER.debug("Found entity %s", name)
+    except PyViCareInternalServerError as server_error:
+        _LOGGER.info(
+            "Server error ( %s): Not creating entity %s", server_error.message, name
+        )
+        return None
+    except PyViCareNotSupportedFeatureError:
+        _LOGGER.info("Feature not supported %s", name)
+        return None
+    except AttributeError:
+        _LOGGER.debug("Attribute Error %s", name)
+        return None
+
+    return ViCareSwitch(
+        name,
+        vicare_api,
+        device_config,
+        description,
+    )
+
+
+async def async_setup_entry(
+        hass: HomeAssistant,
+        config_entry: ConfigEntry,
+        async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Create the ViCare switch entities."""
+    entities = await hass.async_add_executor_job(
+        create_all_entities, hass, config_entry
+    )
+    async_add_entities(entities)
+
+
+def create_all_entities(hass: HomeAssistant, config_entry: ConfigEntry):
+    """Create entities for all devices and their circuits, burners or compressors if applicable."""
+    name = VICARE_NAME
+    entities = []
+
+    for device in hass.data[DOMAIN][config_entry.entry_id][VICARE_DEVICE_CONFIG]:
+        api = device.asAutoDetectDevice()
+
+        for description in SWITCH_DESCRIPTIONS:
+            entity = _build_entity(
+                f"{name} {description.name}",
+                api,
+                device,
+                description,
+            )
+            if entity is not None:
+                entities.append(entity)
+
+    return entities
+
+
+class ViCareSwitch(SwitchEntity):
+    """Representation of a ViCare switch."""
+
+    entity_description: ViCareSwitchEntityDescription
+
+    def __init__(
+            self, name, api, device_config, description: ViCareSwitchEntityDescription
+    ) -> None:
+        """Initialize the switch."""
+        self.entity_description = description
+        self._device_config = device_config
+        self._api = api
+        self._ignore_update_until = datetime.datetime.utcnow()
+        self._state = None
```

Comment:
Please replace `self._state = None` in with `update()` like in the other integrations. This way we can also write tests which assert the state.

See comment below

Reply:
@lukx Can you please have a look at this change request? - I'm using your custom repo at the moment to control the heating of water when the solar panels are producing too much, but seems I can't log-in anymore using your repo. (401: Token missing. Could be unrelated) - But I still think it'd be better to get this integrated in the official repo :)

Thanks!